src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -459,9 +512,10 @@ def parse_args():
         passthrough_index = len(argv)
 
     args = parser.parse_args(argv[1:passthrough_index])
+    args = _check_model_args(parser, args)
     args = _check_conditional_args(parser, args)
+    args = _check_compare_args(compare_parser, args)
     args = _update_load_manager_args(args)
     args = _convert_str_to_enum_entry(args, "prompt_source", PromptSource)
-    args = _prune_args(args)
```

Comment:
Is this not needed?

Reply:
I don't think we need this. And besides, this was causing more workaround because we need to use `hasattr` whenever we needed to check if some argument was set or not. Also, when building PA command, we can ignore all None values easily as well.

Reply:
Thanks for explaining. I'll resolve this. Please do make sure you test end to end since the CI is not up, so I don't know if this would break any edge cases.

CC: @debermudez

Reply:
Yea we are on the same page.
this feels like a change we "think" should be fine, but magically breaks something.
It should be fine if we test it thoroughly.

Reply:
I had the same concern but it doesn't break the e2e runs for trtllm, vllm, and openai endpoints so at least I know all the default runs are working.